extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelper.java

```diff
@@ -1196,4 +1334,47 @@ public static String unescapeDocument(@Nonnull final String doc) {
                 .replaceAll("\\\\x5b", "[")
                 .replaceAll("\\\\x5d", "]");
     }
+
+    /**
+     * Generate a content playback nonce (also called {@code cpn}), sent by YouTube clients in
+     * playback requests (and also for some clients, in the player request body).
+     *
+     * @return a content playback nonce string
+     */
+    @Nonnull
+    public static String generateContentPlaybackNonce() {
+        final SecureRandom random = new SecureRandom();
```

Comment:
Oh, also, I don't know if it is good practice to create a new `SecureRandom` each time, maybe a static field is better?

Reply:
I don't know, sorry.